src/clj_kondo/impl/analyzer/namespace.clj

```diff
@@ -414,6 +414,23 @@
                                     :syntax
                                     "namespace name expected"))))
                  'user)
+        _ (let [expected-file-path (-> (str (munge ns-name))
+                                       (str/replace #"\." "/"))]
+            (when-not (or (= "<stdin>" (:filename ctx))
+                          (= 'user ns-name)
+                          ;; Test for .indexOf because there can be something before
+                          ;; (e.g. 'corpus/' in the tests), and the "expected-file-path"
+                          ;; does not include the extension.
+                          (and (seq (:filename ctx)) (<= 0 (.indexOf ^String (:filename ctx) ^String expected-file-path))))
```

Comment:
Suggestions:

- Pull out `(:filename ctx)` to a let
- Don't call `seq` on the string, just test if the value is there
- Use `str/index-of` rather than interop

Reply:
Another suggestion: just call `(str/index-of (str ...) ...)` on the filename: this should prevent any NPE issues